### ISSUE TITLE
Falco using tokens past their lifetime in Kubernetes 1.21

### ISSUE THREAD

**Issue by plnordquist** [falcosecurity/falco#1781]
**Describe the bug**

In Kubernetes 1.21, the [BoundServiceAccountTokenVolume](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume) feature is enabled by default.  This feature replaces the mounted token secret from the past with a projected volume that includes an auto rotated token.  With a default configuration, the Kubernetes API server monitors when tokens are used past their lifetime and annotates requests in the audit log with a stale-token annotation.  Based on my testing, Falco does not load in rotated tokens and uses the first token mounted in the pod forever.

**How to reproduce it**

1. Deploy falco to Kubernetes with the Helm Chart
2. Wait 1 hour for the token to expire
3. Observe the audit log annotating requests from Falco with the stale-token annotation

**Expected behaviour**

Falco should load the rotated token instead of caching the initial token.

**Environment**

- Falco version: 0.30.0

- System info:
```json
{
  "machine": "x86_64",
  "nodename": "falco-4wlp5",
  "release": "3.10.0-1160.42.2.el7.x86_64",
  "sysname": "Linux",
  "version": "#1 SMP Tue Sep 7 14:49:57 UTC 2021"
}
```

- Cloud provider or hardware configuration:
- OS: CentOS 7

- Kernel: 3.10.0-1160.42.2.el7.x86_64

- Installation method: Kubernetes Helm Chart 

**Additional context**

Example Audit log line:

```json
{
  "kind": "Event",
  "apiVersion": "audit.k8s.io/v1",
  "level": "Metadata",
  "auditID": "a6a52418-b225-43a2-b48b-56e489b9fbab",
  "stage": "RequestReceived",
  "requestURI": "/api/v1/watch/nodes?pretty=false",
  "verb": "watch",
  "user": {
    "username": "system:serviceaccount:services:falco",
    "uid": "146556d9-3753-4f47-922b-cd68f8597d1d",
    "groups": [
      "system:serviceaccounts",
      "system:serviceaccounts:services",
      "system:authenticated"
    ],
    "extra": {
      "authentication.kubernetes.io/pod-name": ["falco-gg6rl"],
      "authentication.kubernetes.io/pod-uid": [
        "760c5c22-73c4-4da0-a21a-ca6e6b57513c"
      ]
    }
  },
  "sourceIPs": ["10.17.2.2"],
  "userAgent": "falcosecurity-libs",
  "objectRef": { "resource": "nodes", "apiVersion": "v1" },
  "requestReceivedTimestamp": "2021-11-10T23:26:08.128787Z",
  "stageTimestamp": "2021-11-10T23:26:08.128787Z",
  "annotations": {
    "authentication.k8s.io/stale-token": "subject: system:serviceaccount:services:falco, seconds after warning threshold: 1388501"
  }
}
```

**Comment by rdubya16** [comment]
Any update on this? We are seeing this as well.

**Comment by ianmuge** [comment]
Any update on this?

**Comment by ashuec90** [comment]
Hello @jasondellaluce Any update on this? We are using eks and we have to upgrade eks 1.20 to 1.21 by end of this month, and in eks 1.21 we need this otherwise we have to restart the falco pods everytime before the token is marked as stale.

Thanks

**Comment by gouthz** [comment]
Hi, Do we have any updates here?

**Comment by Speeddymon** [comment]
Related issue: https://github.com/falcosecurity/falco/issues/2485

**Comment by Andreagit97** [comment]
The new `k8smeta` plugin should solve this issue, give it a try with our dedicated helm chart and Falco 0.36.2 https://github.com/falcosecurity/falco/issues/2973#issuecomment-1876880587

**Comment by Andreagit97** [comment]
This should be solved by [Falco 0.37.0](https://github.com/falcosecurity/falco/releases/tag/0.37.0)! Feel free to reopen if this is still an issue
